ft_statistics_mvpa.m

```diff
@@ -0,0 +1,231 @@
+% Classification using a feature searchlight approach can highlight which
+% feature(s) are informative. To this end, classification is performed on 
+% each feature separately. However, neighbouring features can enter the 
+% classification together when a matrix of size [features x features]
+% specifying the neighbours is provided. Additional parameters:
+% .nb          - [features x features] matrix specifying which features
+%                are neighbours of each other.
+%                          - EITHER - 
+%                a GRAPH consisting of 0's and 1's. A 1 in the 
+%                (i,j)-th element signifies that feature i and feature j 
+%                are neighbours, and a 0 means they are not neighbours
+%                            - OR -
+%                a DISTANCE MATRIX, where larger values mean larger distance.
+%                If no matrix is provided, every feature is only neighbour
+%                to itself and classification is performed for each feature 
+%                separately.
+% .size        - if a nb matrix is provided, size defines the 
+%                size of the 'neighbourhood' of a feature.
+%                if nb is a graph, it gives the number of steps taken 
+%                     through the nb matrix to find neighbours:
+%                     0: only the feature itself is considered (no neighbours)
+%                     1: the feature and its immediate neighbours
+%                     2: the feature, its neighbours, and its neighbours'
+%                     neighbours
+%                     3+: neighbours of neighbours of neighbours etc
+%                     (default 1)
+%                if nb is a distance matrix, size defines the number of
+%                     neighbouring features that enter the classification
+%                     0: only the feature itself is considered (no neighbours)
+%                     1: the feature and its first closest neighbour 
+%                        according to the distance matrix
+%                     2+: the 2 closest neighbours etc.
+%
+% -- TODO: for time x time generalisation, in MVPA light we can use two
+% different datasets (one for training the classifier, the other one for
+% testing). This could be realised eg using extra fields in cfg such as
+% cfg.X2 for dataset 2 and cfg.design2 for the second design matrix
+%
+% Returns:
+%   stat        = struct with results. the .metric field contains the
+%                   requested metrics
+% 
+% (c) matthias treder
+
+% do a sanity check on the input data
+assert(isnumeric(dat),    'this function requires numeric data as input, you probably want to use FT_TIMELOCKSTATISTICS, FT_FREQSTATISTICS or FT_SOURCESTATISTICS instead');
+assert(isnumeric(design), 'this function requires numeric data as input, you probably want to use FT_TIMELOCKSTATISTICS, FT_FREQSTATISTICS or FT_SOURCESTATISTICS instead');
+
+% flip dimensions such that the number of trials comes first
+dat = dat';
```

Comment:
OK, can do that. 

The toolbox has a function /startup/startup_MVPA_Light.m that adds all the folders to the path. The /startup folders contains only this function. Is it possible to contain this folder in the standard FT path? Calling startup_MVPA_Light is more elegant than users having to do something like addpath(genpath(path/to/MVPA/Light))

Reply:
I think it would be more elegant to solve this in ft_hastoolbox, rather than adding a toolbox-specific dedicated startup file. 
For instance, would it work if mvpa_light is treated similar as spm, as in approx. lines 517 of ft_hastoolbox? We could check for mvpa-light, then do addpath(mvpa_light), and then run the startup function (which will be found by matlab, because the main path has been added). Alternatively, if your startup only does addpath(genpath()) then it can be solved here as well I think.

Reply:
This makes sense. In startup_mvpa_light I add the folders by hand because addpath(genpath(...)) would also add all the .git clutter. However, if you include the toolbox in Fieldtrip this wouldn't apply so a simple addpath(genpath(toolbox)) would actually do. I'll have a look..

Reply:
to follow up, obviously the toolbox should exist, should I add it in to fieldtrip/external/ myself? This would probably need another pull request. For the future, when there's a significant update in the toolbox that I'd like to be puished into Fieldtrip I'll keep doing pull requests?

Reply:
Given that mvpa_light is more or less self-contained, and already version controlled in its own repo, I'd suggest not to add it to fieldtrip/external.
Rather, I'd add an entry for mvpa_light to the switch box in ft_hastoolbox, which checks for the presence of a to-be-expected function from the toolbox on the current matlab path.

Next, you should add a line to the 'url' cell-array (starts ~line 60) that points the user to your repository. 

And then we can consider  making an exception for mvpa_light in the myaddpath subfunction (e.g. doing an addpath followed by a call to your startup function).

Yet, what is the problem for adding the .git stuff to the path (apart from it taking some time), so I wonder whether that is needed to begin with?

Reply:
The .git files clutter the namespace and matlab path though it's unlikely to cause any real problems.
I think I don't understand the use of myaddpath then: If mvpa_light is an external toolbox, it's either already in the path (in that case we don't need to do anything and ft_hastoolbox is happy). Or it is not in the path but then we have no idea where to find it since it's not in /external but somewhere else on the computer.

So it's up to the user to correctly install the toolbox isn't it?

Reply:
Uhm, yes you're right. What was I thinking, indeed it's the user's responsibility.
So only we'd need a line in the url cell array, and a check for some of the key functions from the toolbox

Reply:
OK I've added the lines and pushed it. Guess we're done here for now! I just need to add an installation instruction to the tutorial.